tgui/packages/common/color.js

```diff
@@ -65,3 +65,13 @@ Color.lookup = (value, colors = []) => {
   const index = scaled | 0;
   return Color.lerp(colors[index], colors[index + 1], ratio);
 };
+
+// Darkens a colour by a given percent. Only takes hex values. Returns a color, which can have toString called to get it's rgba() css value.
+export const darkenColor = (color, percent) => {
```

Comment:
Would probably work better as a method on the class, so that you could:

```js
Color.fromHex(hex).darken(50)
```

Method has to be immutable though, create a new instance of Color.

Something like this:

```js
return new Color(
  this.r - this.r * percent,
  this.g - this.g * percent,
  this.b - this.b * percent,
  this.a
);
```

Reply:
Done!